GoodreadsEDA/scp_bks.py

```diff
@@ -104,97 +78,124 @@ def print_pg_number(page_count):
+        book_urls ([[str, str, str]]): A list of list of strings consisting of book author, book title, and book url
+          to visit to serve as the base scraping url.
+        driver (webdriver): The selenium chromedriver object to visit the urls with.
+        cursor (sqlite3.cursor): A sqlite cursor to pass to methods writing to the DB.
+    """
+
+    scraping_start_timer = time.perf_counter()
+
+    for book_number, entry in enumerate(book_urls):
         author = entry[0]
         title = entry[1]
-        if '"' in title:
-            title = replace_double_quotes(title)
         url = entry[2]
 
-        print_bk_info(author, title, url)
-
-        driver = webdriver.Chrome(chrome_options=options)
         driver.get(url)
 
-        page_count = 1
         # Goodreads keeps a collection of 10 pages of 30 reviews available on the
         # book's page at a time.
-        for x in range(0, 10):
-            print_pg_number(page_count)
+        print("Scraping 10 pages of reviews for \"{}\". Scraping page ".format(title), end="", flush=True)
+        for page_count in range(1, 11):
+            # print_pg_number(page_count)
+            # each set of 10 pages is printed on the same line to save space & cut down on noise
+            if page_count != 10:
+                print("{}..".format(page_count), end="", flush=True)
+            else:
+                print("{}".format(page_count))
             # grab every review on the current page and put it into "Reviews"
             reviews = driver.find_elements_by_class_name("reviewHeader")
-            for i, review in enumerate(reviews):
+            for review in reviews:
                 user, date, score = get_individual_review(review)
-                insert_into_review_table(c, author, title, url,
-                                         score, user, date)
-                conn.commit()
+                insert_into_review_table(cursor, author, title, url, score, user, date)
 
             link_found = click_next_review_page(driver)
 
             if not link_found or len(reviews) < 30:
                 break
-            else:
-                page_count += 1
-                continue
 
+        print_scraping_status(book_number, book_urls, scraping_start_timer)
+
+
+def main():
+    # Get a list of all the urls for the books by the authors supplied from
+    # mk_bklst.py, and remove the header row.
+    book_urls = get_book_urls()
+
+    # Options for the Chrome driver, so Chrome operates without calling a window.
+    options = webdriver.ChromeOptions()
+    options.add_argument('headless')
+
+    # Connect to db, and create new table "Reviews"
+    conn = sqlite3.connect(f"{os.getcwd()}/review_dbs/reviews.db")
+    c = conn.cursor()
+    create_review_table(c)
+    driver = webdriver.Chrome(chrome_options=options)
+
+    # Making sure the DB and driver connections are closed if there are any errors
+    try:
+        scrape_book_info(book_urls, driver, c)
+    finally:
```

Comment:
A good way to avoid memory leaks and permissions issues is to make sure your various connections are closed whether the script succeeds or not